src/libraries/Microsoft.Extensions.Configuration.Binder/src/ConfigurationBinder.cs

```diff
@@ -687,12 +728,32 @@ private static bool IsArrayCompatibleReadOnlyInterface(Type type)
             return allProperties;
         }
 
+        [RequiresUnreferencedCode(PropertyTrimmingWarningMessage)]
+        private static object? BindParameter(ParameterInfo parameter, IConfiguration config, BinderOptions options)
+        {
+            string parameterName = parameter.Name!;
+
+            var propertyBindingPoint = new BindingPoint(initialValue: config.GetSection(parameterName).Value, isReadOnly: false);
+
+            BindInstance(
+                parameter.ParameterType,
+                propertyBindingPoint,
+                config.GetSection(parameterName),
+                options);
+
+            return propertyBindingPoint.Value;
+        }
+
         private static string GetPropertyName(MemberInfo property)
         {
             ThrowHelper.ThrowIfNull(property);
+            return TryGetNameFromAttributes(property.GetCustomAttributesData()) ?? property.Name;
+        }
 
+        private static string? TryGetNameFromAttributes(IList<CustomAttributeData> customAttributesData)
```

Comment:
(nit) Is this refactoring necessary? I only see this method called from one place. It could be left to the original code.

Reply:
Fair point. As part of the evolution of this PR, it was used in two places. But it's not, so reverted it to how it was.